File: practice/algorithms/implementation/apple_and_orange/AppleAndOrange.scala
import scala.io.Source
import java.util.Scanner

object AppleAndOrange extends App {
  val stdin = new Scanner(System.in)
  val s = stdin.nextInt()
  val t = stdin.nextInt()
  val a = stdin.nextInt()
  val b = stdin.nextInt()
  val m = stdin.nextInt()
  val n = stdin.nextInt()
  val apples = (0 until m).map(_ => stdin.nextInt())
  val oranges = (0 until n).map(_ => stdin.nextInt())
  val fallenApples = apples.count(ap => (a + ap >= s) && (a + ap <= t))
  val fallenOranges = oranges.count(o => (b + o >= s) && (b + o <= t))
	        
  println(fallenApples)
  println(fallenOranges)
}

File: practice/algorithms/implementation/apple_and_orange/AppleAndOrange.java
import java.util.*;

public class AppleAndOrange {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int s = stdin.nextInt();
        int t = stdin.nextInt();
        int a = stdin.nextInt();
        int b = stdin.nextInt();
        int m = stdin.nextInt();
        int n = stdin.nextInt();
        int[] apples = new int[m];
        for(int i = 0; i < m; i++) {
            apples[i] = stdin.nextInt();
        }
        int[] oranges = new int[n];
        for(int i = 0; i < n; i++) {
            oranges[i] = stdin.nextInt();
        }

        int fallenApples = (int) Arrays.stream(apples).filter(ap -> (a + ap >= s) && (a + ap <= t)).count();
        int fallenOranges = (int) Arrays.stream(oranges).filter(o -> (b + o >= s) && (b + o <= t)).count();

        System.out.println(fallenApples);
        System.out.println(fallenOranges);

        stdin.close();
    }
}